src/control/security/pem.go

```diff
@@ -0,0 +1,107 @@
+package security
+
+import (
+	"crypto"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+const (
+	SafeKeyPerm  os.FileMode = 0400
+	SafeCertPerm os.FileMode = 0664
+	SafeDirPerm  os.FileMode = 0700
+)
+
+type insecureError struct {
+	filename string
+	perms    os.FileMode
+}
+
+func (e *insecureError) Error() string {
+	return fmt.Sprintf("%s has insecure permissions set to %#o",
+		e.filename, e.perms)
+}
+
+func LoadPEMData(filePath string, perms os.FileMode) ([]byte, error) {
+	statInfo, err := os.Stat(filePath)
```

Comment:
Is it worth sanity-checking the file size as well to avoid a DOS? Maybe set a maxFileSize const to 1MB or whatever and return an error if statInfo.Size() is larger than that.

Reply:
We don't really do this anywhere else we load up files. We can if you think its important enough but the spec says these files can be any length.

Reply:
Yeah I was just thinking that because this is a security-sensitive area of the code it might warrant extra precautions. I don't feel strongly one way or another; the thought just occurred to me while I was reviewing. This might fall into a wider discussion about hardening.